scripts/build/gen_isr_tables.py

```diff
@@ -172,14 +225,20 @@ def write_address_irq_vector_table(fp, vt, nv):
 #include <zephyr/sw_isr_table.h>
 #include <zephyr/arch/cpu.h>
 
+typedef void (* DIRECT_ISR)();
 typedef void (* ISR)(const void *);
 """
 
-def write_source_file(fp, vt, swt, intlist, syms):
+def write_source_file(fp, vt, swt, intlist, syms, shi, shdi):
     fp.write(source_header)
 
     nv = intlist["num_vectors"]
 
+    if "CONFIG_ARCH_SUPPORT_SHARED_IRQ" in syms:
```

Comment:
The `CONFIG_ARCH_HAS` symbols are not used like this. You have `CONFIG_ARCH_HAS_*` enabled by the architecture and then you have a different symbol `CONFIG_WHATEVER` depending on `CONFIG_ARCH_HAS*` enabled by the SoC / board. Your check must be on `CONFIG_WHATEVER`, not on `CONFIG_ARCH_HAS_*`

Reply:
I've made change